src/components/OrderBookWidget.tsx

```diff
@@ -1,31 +1,180 @@
 import { TokenDetails } from 'types'
+import * as am4core from '@amcharts/amcharts4/core'
+import * as am4charts from '@amcharts/amcharts4/charts'
+import am4themesSpiritedaway from '@amcharts/amcharts4/themes/spiritedaway'
 
 interface OrderBookProps {
   baseToken: TokenDetails
   quoteToken: TokenDetails
+  networkId: number
 }
 
 const Wrapper = styled.div`
   display: flex;
-  align-items: center;
   justify-content: center;
 
-  background-color: #757597;
-  min-height: 25rem;
+  min-height: 40rem;
   color: white;
   text-align: center;
   font-size: 1.6rem;
-  padding: 3rem;
   width: 100%;
 `
 
+enum Offer {
+  Bid,
+  Ask,
+}
+
+interface RawItem {
+  price: number
+  volume: number
+}
+
+interface ProcessedItem {
+  volume: number
+  totalVolume: number
+  askValueY: number | null
+  bidValueY: number | null
+  price: number
+}
+
+const orderbookUrl = (baseToken: TokenDetails, quoteToken: TokenDetails, networkId?: number): string => {
+  let network = 'mainnet'
+  if (networkId === 4) {
+    network = 'rinkeby'
+  }
+  return `https://price-estimate-${network}.dev.gnosisdev.com/orderbook?base=${baseToken.id}&quote=${quoteToken.id}`
+}
+
+/**
+ * This method turns the raw data that the backend returns into data that can be displayed by the chart.
+ * This involves aggregating the total volume and accounting for decimals
+ */
+const processData = (
+  list: RawItem[],
+  baseToken: TokenDetails,
+  quoteToken: TokenDetails,
+  type: Offer,
+): ProcessedItem[] => {
+  let totalVolume = 0
+  return (
+    list
+      // Account fo decimals
+      .map(element => {
+        return {
+          price: element.price / 10 ** (quoteToken.decimals - baseToken.decimals),
+          volume: element.volume / 10 ** baseToken.decimals,
+        }
+      })
+      // Filter tiny orders
+      .filter(e => e.volume > 0.01)
+      // Accumulate totalVolume
+      .map(e => {
+        const previousTotalVolume = totalVolume
+        totalVolume += e.volume
+        return {
+          price: e.price,
+          volume: e.volume,
+          totalVolume,
+          // Amcharts draws step lines so that the x value is centered (Default). To correctly display the orderbook, we want
+          // the x value to be at the left side of the step for asks and at the right side of the step for bids.
+          //
+          //    Default            Bids          Asks
+          //            |      |                        |
+          //   ---------       ---------       ---------
+          //  |                         |      |
+          //       x                    x      x
+          //
+          // For asks, we can offset the "startLocation" by 0.5. However, Amcharts does not support a "startLocation" of -0.5.
+          // For bids, we therefore offset the curve by -1 (expose the previous total volume) and use an offset of 0.5.
+          // Otherwise our steps would be off by one.
+          askValueY: type == Offer.Bid ? previousTotalVolume : null,
```

Comment:
This can be moved up to a `isBid` const so you don't reevaluate for every order
the `isAsk` can be defined as `!isBid` right?